sdk/core/azure-core/inc/azure/core/internal/extendable_enumeration.hpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief Internal utility functions for extendable enumerations.
+ *
+ */
+#include <string>
+
+namespace Azure { namespace Core { namespace _internal {
+  /** @brief Template base class helper for implementing extendable enumerations.
+   *
+   * This template exists to simplify the experience of authoring ["extendable
+   * enumerations"](https://azure.github.io/azure-sdk/cpp_implementation.html#cpp-enums).
+   *
+   * An extendable enumeration derives from the #ExtendableEnumeration base class passing in the
+   * extendable enumeration type as the template specialization.
+   *
+   * Example:
+   *
+   * \code{.cpp}
+   * class MyEnumeration final : public ExtendableEnumeration<AttestationType> {
+   * public:
+   *   MyEnumeration(std::string attestationType) :
+   *     ExtendableEnumeration(std::move(attestationType)) {}
+   *   MyEnumeration() = default;
+   *   static const MyEnumeration Enumerator1;
+   *   static const MyEnumeration Enumerator2;
+   *   static const MyEnumeration Enumerator3;
+   * };
+   * \endcode
+   *
+   */
+  template <class T> class ExtendableEnumeration {
+  private:
+    std::string m_enumerationValue;
+
+  public:
+    /**
+     * @brief Construct a new extensable enumeration object
+     *
+     * @param enumerationValue The string enumerationValue used for the value.
+     */
+    explicit ExtendableEnumeration(std::string enumerationValue)
+        : m_enumerationValue(std::move(enumerationValue))
+    {
+    }
+
```

Comment:
I suggest adding a protected defaulted destructor:
```c++
protected:
  ~Extendable() = default;
```
here somewhere to avoid folks inadvertently destroying an object of the derived type via a pointer or reference to `ExtendableEnumeration`. (On high enough settings, compilers will warn about deriving from a class that doesn't have a virtual or protected destructor.)

Reply:
Sounds good.